include/solvers/navier_stokes_assemblers.h

```diff
@@ -338,4 +338,47 @@ class LaplaceAssembly : public NavierStokesAssemblerBase<dim>
 };
 
 
+/**
+ * @brief Class that assembles the Buoyancy using Boussinesq approximation
+ *
+ * @tparam dim An integer that denotes the number of spatial dimensions
+ *
+ * @ingroup assemblers
+ */
+
+template <int dim>
+class BuoyancyAssembly : public NavierStokesAssemblerBase<dim>
+{
+public:
+  BuoyancyAssembly(std::shared_ptr<SimulationControl> simulation_control,
+                   Parameters::PhysicalProperties     physical_properties)
+    : simulation_control(simulation_control)
+    , physical_properties(physical_properties)
+  {}
+
+  /**
+   * @brief assemble_matrix Assembles the matrix
+   * @param scratch_data (see base class)
+   * @param copy_data (see base class)
+   */
+  virtual void
+  assemble_matrix(NavierStokesScratchData<dim> &        scratch_data,
+                  StabilizedMethodsTensorCopyData<dim> &copy_data) override;
+
+
+  /**
+   * @brief assemble_rhs Assembles the rhs
```

Comment:
Can you give a bit more detail? What is the term being assembled exactly?

Reply:
Added.